src/components/DeliverableView/DocumentsUploader.tsx

```diff
@@ -23,21 +38,29 @@ const DocumentsUploader = ({ deliverable }: ViewProps): JSX.Element => {
 
   const onCloseFileUploadDialog = useCallback(() => void setFiles([]), []);
 
+  useEffect(() => {
+    if (!canEdit && documentLimitReached && showDocumentLimitReachedMessage) {
```

Comment:
We don't want to show this message for READ_ONLY internal users.
The canEdit property coming from AcceleratorRouters captures which global roles can upload files.
I added a comment in DeliverableRouter/DeliverableView suggesting we isolate logic to show the message and hide the documents uploader locally - similar to how we hide the DocumentsUploader in AcceleratorRouter if user does not have edit permissions.
We can keep this component clean of changes except maybe the MAX_FILES_LIMIT change.